bin/server/cmd/root.go

```diff
@@ -131,10 +124,19 @@ func Execute() error {
 }
 
 func init() {
-	rootCmd.PersistentFlags().StringVar(&flags.host, "host", "http://localhost", "host where Bytebase backend is accessed from, must start with http:// or https://. This is used by Bytebase to create the webhook callback endpoint for VCS integration")
-	rootCmd.PersistentFlags().IntVar(&flags.port, "port", 80, "port where Bytebase backend is accessed from. This is also used by Bytebase to create the webhook callback endpoint for VCS integration")
-	rootCmd.PersistentFlags().StringVar(&flags.frontendHost, "frontend-host", "", "host where Bytebase frontend is accessed from, must start with http:// or https://. This is used by Bytebase to compose the frontend link when posting the webhook event. Default is the same as --host")
-	rootCmd.PersistentFlags().IntVar(&flags.frontendPort, "frontend-port", 0, "port where Bytebase frontend is accessed from. This is used by Bytebase to compose the frontend link when posting the webhook event. Default is the same as --port")
+	// In the release build, Bytebase runs as a mono server bundling frontend and backend together. Thus --host:--port is also where frontend runs.
+	// During development, Bytebase frontend runs on a separate port.
+	rootCmd.PersistentFlags().StringVar(&flags.host, "host", "0.0.0.0", "host where Bytebase backend runs. Default to 0.0.0.0")
+	rootCmd.PersistentFlags().IntVar(&flags.port, "port", 80, "port where Bytebase backend runs. Default to 80")
+	// When running the release build in production, most of the time, users would not expose the --host:--port directly to the public.
+	// Instead they would configure a gateway to forward the traffic to --host:--port. Users need to set --external-url to the address
+	// exposed on the gateway accordingly.
+	//
+	// It's important to set the correct --external-url. This is used for:
+	// 1. Constructing the correct callback URL when configuring the VCS provider. The callback URL points to the frontend.
+	// 2. Creating the correct webhook endpoint when configuring the project GitOps workflow. The webhook endpoint points to the backend.
+	// Since frontend and backend are bundled and run on the same address in the release build, thus we just need to specify a single external URL.
+	rootCmd.PersistentFlags().StringVar(&flags.externalURL, "external-url", "", "the external URL where user visits Bytebase, must start with http:// or https://. Default value is based on --host --port")
```

Comment:
Shall we require the --external-url flag regardless? We can set the default value to "http://www.bytebase.com/docs" and have our website "http://www.bytebase.com/docs/oauth/callback" to return the right setup documentation if it works.

Reply:
Or exit with doc link if empty.